ai_economist/foundation/env_wrapper.py

```diff
@@ -94,48 +96,59 @@ class FoundationEnvWrapper:
     def __init__(
         self,
         env_obj=None,
+        env_name=None,
+        env_config=None,
         num_envs=1,
         use_cuda=False,
-        customized_env_registrar=None,
+        testing_mode=False,
+        env_registry=None,
+        event_messenger=None,
+        process_id=0,
     ):
         """
-        'env_obj': an environment instance
+        'env_obj': an environment object
+        'env_name': an environment name that is registered on the
+            WarpDrive environment registrar
+        'env_config': environment configuration to instantiate
+            an environment from the registry
         'use_cuda': if True, step through the environment on the GPU, else on the CPU
         'num_envs': the number of parallel environments to instantiate. Note: this is
-        only relevant when use_cuda is True
-        'customized_env_registrar': CustomizedEnvironmentRegistrar object
+            only relevant when use_cuda is True
+        'testing_mode': a flag used to determine whether to simply load the .cubin (when
+            testing) or compile the .cu source code to create a .cubin and use that.
+        'env_registry': EnvironmentRegistrar object
             it provides the customized env info (like src path) for the build
-            on a GPU (when use_cuda is True)
+        'event_messenger': multiprocessing Event to sync up the build
+            when using multiple processes
+        'process_id': id of the process running WarpDrive
         """
         # Need to pass in an environment instance
-        assert env_obj is not None
-        self.env = env_obj
+        if env_obj is not None:
+            self.env = env_obj
+        else:
+            assert (
+                env_name is not None
+                and env_config is not None
+                and env_registry is not None
+            )
+            self.env = env_registry.get(env_name, use_cuda)(**env_config)
+        # testing mode should always be False here,
+        # as we will compile the .cu source code to create a .cubin
+        assert not testing_mode
```

Comment:
if testing_mode is True, nothing meaningful happens. remove testing_mode, just make a dummy Env to support testing if necessary. Also, testing_mode -> debug_mode?

Reply:
Agreed. I removed testing_mode as it's not relevant here.